sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -490,6 +490,24 @@ def test_spline_transformer_n_features_out(
     assert splt.transform(X).shape[1] == splt.n_features_out_
 
 
+def test_spline_transformer_with_constant_and_near_constant_features(
+    global_random_seed,
+):
+    """Test SplineTransformer does not output nan values for constant
+    and near constant features."""
+    rng = np.random.RandomState(global_random_seed)
+
+    spt = SplineTransformer(extrapolation="periodic")
```

Comment:
We might want to parametrize this test on `extrapolation` to increase coverage for this edge case.